rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "fastrtps/rtps/common/Guid.h"
+
+#include "rmw_fastrtps_shared_cpp/guid_utils.hpp"
+
+using rmw_fastrtps_shared_cpp::copy_from_byte_array_to_fastrtps_guid;
+using rmw_fastrtps_shared_cpp::copy_from_fastrtps_guid_to_byte_array;
+
+static constexpr size_t byte_array_size =
+  eprosima::fastrtps::rtps::GuidPrefix_t::size +
+  eprosima::fastrtps::rtps::EntityId_t::size;
+
+TEST(GUIDUtilsTest, byte_array_to_guid_and_back) {
+  uint8_t input_byte_array[byte_array_size] = {0};
+  input_byte_array[0] = 0xA5;
+  input_byte_array[byte_array_size - 1] = 0x4B;
+  eprosima::fastrtps::rtps::GUID_t guid;
+  copy_from_byte_array_to_fastrtps_guid(input_byte_array, &guid);
```

Comment:
While converting back and forth between the two does ensure that no data is lost, it doesn't really test that it was converted to the correct output. It is also possible that this won't catch errors if they are introduced in the header if somehow both methods correct each other's mistake.

Reply:
That is true, I just didn't want to pollute the test with too much information as to how it gets copied in-n-out.